code/include/back-end/lexer.hpp

```diff
@@ -3,32 +3,30 @@
 #include "token.hpp"
 
 #include <fstream>
-#include <optional>
 #include <string>
 #include <vector>
-#include <map>
 #include <unordered_map>
 
 // one lexer for each file
 class Lexer
 {
 public:
-	Lexer(std::string_view file_name, bool main_file);
+	Lexer(std::string_view file_name, const bool main_file);
 
 public:
-	Token eat(bool go_to_next_line);
-	Token peek(bool go_to_next_line);
+	Token eat(const bool go_to_next_line);
+	Token peek(const bool go_to_next_line) noexcept;
```

Comment:
`peek` can throw because it calls `eat` which does throw

Reply:
Oh yeah, that one's actually a mistake since that would probably immediately call `std::terminate` if `eat` errors whilst being called by `peek`. My bad :/